test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -656,6 +657,121 @@ def closure_sharded(input_tensor=input_tensor):
             for opt in [torch.optim.SGD, torch.optim.Adam]:
                 check_optimizer_equivalence(opt)
 
+    @common_distributed.skip_if_lt_x_gpu(2)
+    def test_zero_join(self):
```

Comment:
(can be done in followup PR) shall we also test CPU-only training?

Reply:
Added a CPU-only test using `gloo` backend.